### PR TITLE
fix: styling issues with the Popover component within the Space.Compa…   

### PR DESCRIPTION
**Pull request by foryuki**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/41673

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix styling issues with components that have a Tooltip component as their base within the Space.Compact wrapper.      |
| 🇨🇳 Chinese |     修复 Space.Compact 包裹底层为 Tooltip 组件的组件时的样式问题。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 61aef13</samp>

Fixed a visual issue with popovers in compact space style. Applied `NoCompactStyle` to `Tooltip` in `components/popover/index.tsx`.

### <samp>🤖 Generated by Copilot at 61aef13</samp>

*  Wrap `Tooltip` component with `NoCompactStyle` to fix popover spacing bug ([link](https://github.com/ant-design/ant-design/pull/41707/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0R4), [link](https://github.com/ant-design/ant-design/pull/41707/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0L58-R74))
*  Import `NoCompactStyle` component from `../space/Compact` ([link](https://github.com/ant-design/ant-design/pull/41707/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0R4))

### CODE REVIEW

**Comment by foryuki** [comment]
同遇到这个问题 https://github.com/ant-design/ant-design/pull/41701/files

**Comment by foryuki** [comment]
tooltip snapshot 里为啥会有这种自闭合的 div 呢？https://github.com/ant-design/ant-design/actions/runs/4645420271/jobs/8221277920?pr=41707
Screenshot 2023-04-08 at 10 26 17 PM

**Comment by zombieJ** [comment]
我先 merge 一下 master

File: components/tooltip/index.tsx [hunk 1]
```diff
@@ -396,7 +397,11 @@ const Tooltip = React.forwardRef<TooltipRef, TooltipProps>((props, ref) => {
       getTooltipContainer={getPopupContainer || getTooltipContainer || getContextPopupContainer}
       ref={tooltipRef}
       builtinPlacements={tooltipPlacements}
-      overlay={memoOverlay}
+      overlay={
+        <NoCompactStyle>
+          {typeof memoOverlay === 'function' ? memoOverlay() : memoOverlay}
```
Comment (zombieJ) [inline components/tooltip/index.tsx:402]: 推测应该是这个不等价了，原本传入 function 是在弹出时执行。现在变成了渲染时执行，这里也需要包一下。如果是个 function 传入的也是个 function 才行

**Comment by zombieJ** [comment]
感觉又是一个 edge case... 17 和 18 表现不一样。

File: components/space/__tests__/space-compact.test.tsx [hunk 2]
```diff
@@ -280,4 +280,37 @@ describe('Space.Compact', () => {
         ?.classList.contains('ant-picker-compact-item'),
     ).toBe(false);
   });
+
+  it('Tooltip content supports function ', () => {
```
Comment (afc163) [inline components/space/__tests__/space-compact.test.tsx:284]:
```suggestion
  it('Tooltip content supports function', () => {
```

**Review by zombieJ — APPROVED** [review]
